From 4d806441483973507ec0f7ccdb7896389058adc0 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 2 Oct 2024 15:30:24 +0200
Subject: [PATCH 1276/1695] dts: stm32: default disable DMA at SoC level for
 stm32mp15 platforms

DMA node in stm32mp15* SoC DTSI files shouldn't be enabled by default,
we don't even have a driver to handle it. Therefore default disable it.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b84c262342f1306b93a70eb058db1c6c9fbd997c dts: stm32: default disable DMA at SoC level for stm32mp15 platforms]
Change-Id: Id83601d7e61d6911fe6b723a6f25c8dc6378a6b2
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/409655
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp151.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp151.dtsi b/core/arch/arm/dts/stm32mp151.dtsi
index 36de0fb0b..e35326445 100644
--- a/core/arch/arm/dts/stm32mp151.dtsi
+++ b/core/arch/arm/dts/stm32mp151.dtsi
@@ -1626,6 +1626,7 @@
 				st,mem2mem;
 				dma-requests = <8>;
 				access-controllers = <&etzpc STM32MP1_ETZPC_DMA1_ID>;
+				status = "disabled";
 			};
 
 			dma2: dma-controller@48001000 {
@@ -1645,6 +1646,7 @@
 				st,mem2mem;
 				dma-requests = <8>;
 				access-controllers = <&etzpc STM32MP1_ETZPC_DMA2_ID>;
+				status = "disabled";
 			};
 
 			dmamux1: dma-router@48002000 {
@@ -1657,6 +1659,7 @@
 				clocks = <&rcc DMAMUX>;
 				resets = <&rcc DMAMUX_R>;
 				access-controllers = <&etzpc STM32MP1_ETZPC_DMAMUX_ID>;
+				status = "disabled";
 			};
 
 			adc: adc@48003000 {
-- 
2.47.2

